include/flatbuffers/idl.h

```diff
@@ -59,6 +59,21 @@ namespace flatbuffers {
   TD(STRUCT, "",       Offset<void>, int, int, int, int) \
   TD(UNION,  "",       Offset<void>, int, int, int, int)
 
+#define FLATBUFFERS_GEN_TYPES_SCALAR_ALIAS(TD) \
```

Comment:
The fact that it requires the macro to be changed is unfortunate, but it is a simple change.

I do think it's the cleaner way, since the macro has one IDLTYPE parameter, and now we'd simply have two. The parser can then check for both in the same loop.